EPFIngester.py

```diff
@@ -74,9 +74,19 @@ def __init__(self,
         """
         self.filePath = filePath
         self.fileName = os.path.basename(filePath)
-        pref = ("%s_" % tablePrefix if tablePrefix else "")
-        self.tableName = (pref + self.fileName).replace("-", "_") #hyphens aren't allowed in table names
+        self.tableName = self.fileName.replace("-", "_") #hyphens aren't allowed in table names
         self.tableName = self.tableName.split(".")[0]
+
+        pref = ""
+        if tablePrefix:
+            # add _ separator to prefix unless the prefix is a schema or empty
+            pref = ("%s_" % tablePrefix if len(tablePrefix) and tablePrefix[-1] != "." else tablePrefix)
+
+        if pref.split(".")[0] != pref:
+            # looks like there's a schema.
+            self.tableSchema, pref = pref.split(".")
```

Comment:
`self.tableSchema` doesn't look to be defined at all if this branch is missed.

Reply:
that's fine; the search_path checks if it's set.

Reply:
FIxed in 4666799